web/src/page_params.ts

```diff
@@ -120,6 +120,7 @@ export const page_params: {
     realm_plan_type: number;
     realm_private_message_policy: number;
     realm_push_notifications_enabled: boolean;
+    realm_rendered_description?: string;
```

Comment:
I'm pretty sure this is always a string.

Reply:
```py
    if user_profile is None:
        # Get rendered version of realm description which is displayed in right
        # sidebar for spectator.
        page_params["realm_rendered_description"] = get_realm_rendered_description(realm)
        page_params["language_cookie_name"] = settings.LANGUAGE_COOKIE_NAME
```

I think it's only added in certain circumstances. I was also able to reproduce having `undefined` locally, though I can double check that.

Reply:
```js
export function initialize_right_sidebar(): void {
    console.log("is it defined?", page_params.realm_rendered_description)
```

Confirmed that it can be undefined.

Reply:
What's the situation where it be undefined? Seems like you're working around some other bug -- or maybe just missing the fact that it's been moved to `state_data.realm`.

Reply:
Given the code block above, it seems like it's `undefined` except for when `user_profile` is `None`.

On main I see this:

```js
export function initialize_right_sidebar() {
    const rendered_sidebar = render_right_sidebar({
        realm_rendered_description: page_params.realm_rendered_description,
    });
```

Should it be `state_data.realm.realm_rendered_description` instead?

Reply:
```js
    console.log(
        page_params.realm_rendered_description,
        realm.realm_rendered_description,
    )
```

gives me `undefined undefined` when logged in, and `<p>The Zulip development environment default organization.  It's great for testing!</p> undefined` when logged out.

Reply:
I think I'd just misread what you were quoting above -- which is that this isn't sent for non-spectators, because we don't display that to them. Possibly we should change that, but it isn't free to compute, so that's probably why we did it that way.